src/extendStyles.jsx

```diff
@@ -0,0 +1,46 @@
+/* eslint react/forbid-foreign-prop-types: off */
+
+import deepmerge from 'deepmerge';
+
+// Recursively iterate through the style object, validating the same path exists in the
+// extendableStyles object.
+function validateStyle(style, extendableStyles, path = '') {
+  if (process.env.NODE_ENV !== 'production') {
+    // Stop recursively validating when we hit a style's value
+    if (typeof style === 'string' || typeof style === 'number') {
```

Comment:
hmm, can a style's value *be* an array'? maybe this check should just be `if (!style || Array.isArray(style) || typeof style !== 'object')`? (more or less what you had before)

Reply:
I wasn't sure if an array was a valid value either. I did a [quick search](https://sourcegraph.d.musta.ch/search?q=/withStyles%5C%28%5B%5CS%5Cs%5D%2B:+%5C%5B%5B%5CS%5Cs%5D%2B%5C%7D%5C%29/+repo:%5Eairbnb/airbnb%24+) but couldn't find any instances of an array being used as a style's value. But this may not account for all usages of `react-with-styles`, I'll update to the above!